src/lib/widget/_utilities/component.utilities.ts

```diff
@@ -0,0 +1,88 @@
+import { writable, type Writable } from 'svelte/store';
+
+import type { SvelteComponent } from 'svelte';
+
+export interface ComponentStoreValue {
+  error: { code: string; message: string } | null;
+  modal: {
+    component: SvelteComponent;
+    element: HTMLDialogElement;
+  } | null;
+  mounted: boolean;
+  open: boolean | null;
+  reason: 'auto' | 'external' | 'user' | null;
+  type: 'inline' | 'modal' | null;
+}
+
+export const componentStore: Writable<ComponentStoreValue> = writable({
+  modal: null,
+  form: null,
+  error: null,
+  mounted: false,
+  open: null,
+  reason: null,
+  type: null,
+});
+
+export const componentApi = () => {
```

Comment:
This replaces the `modal` API from the form factor specific Svelte components. Plus, since both modal and inline share the same `mount` method, I could combine the two APIs.